File: client/src/app/shared/models/product.ts
import { ProductBrand } from "./ProductBrand";
import { ProductType } from "./ProductType";



export interface Product {
  id :number;
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
  productBrand:string;
  productType:string;
  //productBrand:ProductBrand;
  //productType: ProductType;
}

File: Core/Entities/Product.cs
namespace Core.Entities;


public class Product : BaseEntity
{
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public string PictureUrl { get; set; } // New property for the picture URL

    // Foreign key properties
    public int ProductTypeId { get; set; }
    public int ProductBrandId { get; set; }

    // Navigation properties
    public ProductBrand ProductBrand { get; set; }
    public ProductType ProductType { get; set; }
}
